src/spf_checker.rs

```diff
@@ -126,8 +146,134 @@ impl SpfChecker {
             visited: visited.len(),
             spf_record: root_spf_record,
             included_domains: Some(included_domains),
+            fallback_check: false,
+        })
+    }
+
+    async fn check_target_mechanisms(&self, root_domain: &String, target: &String, initial_result: CheckResult) -> Result<CheckResult> {
+        // Resolve the target include's SPF record
+        let Some(target_spf_txt) = self.resolver.find_spf_record(target).await? else {
+            log_message(format!("No SPF record found for target domain: {}", target));
+            return Ok(CheckResult {
+                found: false,
+                visited: initial_result.visited,
+                spf_record: initial_result.spf_record,
+                included_domains: initial_result.included_domains,
+                fallback_check: true,
+            });
+        };
+
+        let target_spf = Spf::from_str(&target_spf_txt).context("TARGET_SPF_PARSE_FAILED")?;
+
+        // Extract mechanisms from target SPF (excluding 'all' mechanisms)
+        let target_mechanisms: Vec<String> = target_spf
+            .iter()
+            .filter(|mechanism| !mechanism.kind().is_all()) // Exclude 'all' mechanisms
+            .map(|mechanism| mechanism.raw())
+            .collect();
+
+        if target_mechanisms.is_empty() {
+            log_message(format!("No mechanisms found in target SPF record: {}", target_spf_txt));
```

Comment:
I think my code editor applied code formation automatically. Maybe you need to run `cargo fmt --all` on your side?

```suggestion
            log_message(format!(
                "No mechanisms found in target SPF record: {target_spf_txt}"
            ));
```